src/main/java/com/griddynamics/gridu/phonebookmvc/service/PhonebookServiceIml.java

```diff
@@ -0,0 +1,76 @@
+package com.griddynamics.gridu.phonebookmvc.service;
+
+import com.griddynamics.gridu.phonebookmvc.entity.Contact;
+import com.griddynamics.gridu.phonebookmvc.repository.ContactRepository;
+import com.griddynamics.gridu.phonebookmvc.exception.ContactAlreadyExistException;
+import com.griddynamics.gridu.phonebookmvc.exception.ContactNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PhonebookServiceIml implements  PhonebookService {
+
+    @Autowired
+    private ContactRepository repository;
+
+    @Override
+    public Contact saveContact(Contact contact) throws ContactAlreadyExistException {
+        try {
+            getContactByName(contact.getName());
+        } catch (ContactNotFoundException e) {
+            return repository.save(contact);
+        }
+        throw new ContactAlreadyExistException(contact.getName());
+    }
+
+    @Override
+    public List<Contact> saveAllContacts(List<Contact> contacts) throws ContactAlreadyExistException {
+        for (Contact contact : contacts) {
+            try {
```

Comment:
Well, the logic looks a bit too heavy. I believe there could be a better way to save a collection. We are not alone, many people solved such a task many times before us.

Reply:
Same as in saveContact(), I use doesContactExist(Contact contact) to check if any member of collection is already exist in DB.